Library/Homebrew/cmd/audit.rb

```diff
@@ -827,12 +827,26 @@ def audit_version
   def audit_checksum
     return unless checksum
 
-    case checksum.hash_type
-    when :md5
-      problem "MD5 checksums are deprecated, please use SHA1 or SHA256"
-      return
-    when :sha1   then len = 40
-    when :sha256 then len = 64
+    strict = ARGV.include? "--strict"
+
+    if strict
+      case checksum.hash_type
+      when :md5
+        problem "MD5 checksums are deprecated, please use SHA256"
+        return
+      when :sha1
+        problem "SHA1 checksums are deprecated, please use SHA256"
+        return
+      when :sha256 then len = 64
+      end
+    else
+      case checksum.hash_type
+      when :md5
+        problem "MD5 checksums are deprecated, please use SHA256"
```

Comment:
Can we still use ` when ` in that situation, or does that need to change to something like:

```ruby
if :sha1
  then len = 40
end

if :sha256
  then len = 64
end
```

And so on?